tangos/input_handlers/pynbody.py

```diff
@@ -433,22 +434,130 @@ def _is_able_to_load(self, filepath):
+        self._f1dm = f1.dm
+        self._f2dm = f2.dm
+
+        return pynbody.bridge.OrderBridge(self._f1dm, self._f2dm, monotonic=False)
+
+    def match_objects(self, ts1, ts2, halo_min, halo_max, dm_only=True, threshold=0.005,
+                      object_typetag="halo", output_handler_for_ts2=None):
+        import pynbody
+        if not dm_only:
+            logger.warning(
+                "`match_objects` was called with dm_only=%s, but %s only supports DM-only"
+                " catalogues at the moment. Falling back to DM-only.", dm_only, self.__class__.__name__
+            )
+            dm_only = True
+
+        return super().match_objects(
+            ts1,
+            ts2,
+            halo_min,
+            halo_max,
+            dm_only=dm_only,
+            threshold=threshold,
+            object_typetag=object_typetag,
+            output_handler_for_ts2=output_handler_for_ts2,
+            fuzzy_match_kwa={"use_family": pynbody.family.dm}
+        )
 
-class GadgetAHFInputHandler(PynbodyInputHandler):
-    patterns = ["snapshot_???"]
-    auxiliary_file_patterns = ["*.AHF_particlesSTARDUST"]
 
-    def _is_able_to_load(self, filepath):
-        try:
-            f = pynbody.load(filepath)
-            h = pynbody.halo.AHFCatalogue(f)
-            return True
-        except (OSError, RuntimeError):
-            return False
+class AHFInputHandler(PynbodyInputHandler):
+    pynbody_halo_class_name = "AHFCatalogue"
+
+    _included_additional_properties = (
+        "parent", "child", "shrink_center", "bulk_velocity",
+    )
+
+    _excluded_precalculated_properties = (
+        "boxsize", "time", "hostHalo", "Xc", "Yc", "Zc",
+        "VXc", "VYc", "VZc",
+    )
+
+    def available_object_property_names_for_timestep(self, ts_extension, object_typetag):
+        h = self._construct_halo_cat(ts_extension, object_typetag)
+
+        return (
+            [
+                key for key in h[1].properties.keys()
+                if key not in self._excluded_precalculated_properties
+            ] + list(self._included_additional_properties)
+        )
 
 
+    def _get_map_child_subhalos(self, ts_extension):
+        h = self._construct_halo_cat(ts_extension, 'halo')
+        halo_children = defaultdict(list)
+        for halo in h:
+            iparent = halo.properties["hostHalo"]
+            if iparent == 0:
+                continue
+
+            halo_children[iparent].append(halo.properties["halo_id"])
+        return halo_children
+
+    def iterate_object_properties_for_timestep(self, ts_extension, object_typetag, property_names):
+        h = self._construct_halo_cat(ts_extension, object_typetag)
+
+        h.physical_units()
+
+        if "child" in property_names:
+            map_child_parent = self._get_map_child_subhalos(ts_extension)
+
+        for halo in h:
+            # Since PR88, Tangos has a finder_offset mapping the index of the halo in the pynbody catalog,
```

Comment:
Is this comment actually needed? It seems like it is covered by the docstring for the base class